packages/astro/src/types/public/config.ts

```diff
@@ -1453,6 +1453,46 @@ export interface AstroUserConfig {
 		 *   }
 		 * })
 		 * ```
+		 * 
+		 * `envField` currently supports 4 data types: string, number, enum and boolean. `context` and `access` are required for all data types. Certain data types have optional and more required options:
```

Comment:
This should have its own entry, with its own version, description tags etc.!

And being its own thing, needs a **definition**.

> Allows you to specify the data type and properties of an environment variable.
>
> `envField` supports four data types: string, number, enum, and boolean. `context` and `access` are required properties for all data types. The following shows the complete list of properties available for each data type: